registrations/views.py

```diff
@@ -125,13 +126,21 @@ def post(self, request, *args, **kwargs):
             )
             return redirect(reverse_lazy("registrations:confirm-clinic"))
 
+        send_registration_to_rapidpro.delay(
+            contact=session["contact"],
+            msisdn=session["registration_details"]["msisdn"],
+            referral_msisdn=session.get("registered_by"),
+            channel=session["channel"],
+            clinic_code=session["registration_details"]["clinic_code"],
+            timestamp=datetime.utcnow().timestamp(),
+        )
```

Comment:
Instead of just firing these both off, I think we can `chain` these two tasks together, that way we only send the registration to openhim if the submission to RapidPro is successful.

http://docs.celeryproject.org/en/latest/userguide/canvas.html#chains

Reply:
@rudigiesler So, trying to call these like this 
```
result = chain(
            send_registration_to_rapidpro.s(...),
            send_registration_to_openhim.s(...)
)
result.get()
```
results in them not being called at all when I run the tests

Reply:
My other concern about this is where it says "The linked task will be applied with the result of its parent task as the first argument", If I understand that correctly then I'll have to change the openhim method to accept a task result as it's first argument? is that correct?

Reply:
I yes, I forgot about that part of chain. So yes, either ignore that argument in `send_registration_to_openhim`, or we can maybe not use `chain` and instead queue up the second task from within the first task, although that seems less clean.

I think you need to do a `.get` on it for some strange reason, looking at my past code: https://github.com/praekeltfoundation/ndoh-hub/blob/e07c87e14e1f6d064675d2928bdc4f5820025358/changes/test_tasks.py#L349 . So what I've done is do the call to the task with the `.get` in the test, testing the task directly, and then when testing the view, just using a mock: https://github.com/praekeltfoundation/ndoh-hub/blob/731dca81d8cea50cc782b7b21c803973e719d87e/changes/test_views.py#L156